contracts/controllers/DelegatedController.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.5.17;
+
+import "@openzeppelinV2/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelinV2/contracts/math/SafeMath.sol";
+import "@openzeppelinV2/contracts/utils/Address.sol";
+import "@openzeppelinV2/contracts/token/ERC20/SafeERC20.sol";
+
+import "../../interfaces/yearn/Strategy.sol";
+import "../../interfaces/yearn/Converter.sol";
+import "../../interfaces/yearn/OneSplitAudit.sol";
+import "../../interfaces/yearn/DelegatedVault.sol";
+
+contract DelegatedController {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public governance;
+    address public onesplit;
+    address public rewards;
+
+    // Vault to strategy mapping
+    mapping(address => address) public vaults;
+    // Strategy to vault mapping
+    mapping(address => address) public strategies;
+
+    mapping(address => mapping(address => address)) public converters;
+
+    mapping(address => bool) public isVault;
+    mapping(address => bool) public isStrategy;
+
+    uint256 public split = 500;
+    uint256 public constant max = 10000;
+
+    constructor(address _rewards) public {
+        governance = msg.sender;
+        onesplit = address(0x50FDA034C0Ce7a8f7EFDAebDA7Aa7cA21CC1267e);
+        rewards = _rewards;
+    }
+
+    function setSplit(uint256 _split) external {
+        require(msg.sender == governance, "!governance");
+        split = _split;
+    }
+
+    function setOneSplit(address _onesplit) external {
+        require(msg.sender == governance, "!governance");
+        onesplit = _onesplit;
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setConverter(
+        address _input,
+        address _output,
+        address _converter
+    ) external {
+        require(msg.sender == governance, "!governance");
+        converters[_input][_output] = _converter;
+    }
+
+    function setStrategy(address _vault, address _strategy) external {
+        require(msg.sender == governance, "!governance");
+        address _current = strategies[_vault];
+        if (_current != address(0)) {
+            Strategy(_current).withdrawAll();
+        }
+        strategies[_vault] = _strategy;
+        isStrategy[_strategy] = true;
+        vaults[_strategy] = _vault;
+        isVault[_vault] = true;
+    }
+
+    function want(address _vault) external view returns (address) {
+        return Strategy(strategies[_vault]).want();
+    }
+
+    function earn(address _vault, uint256 _amount) public {
```

Comment:
Anyone is allowed to up the capital a Strategy has? This seems bad (even though there's no direct theft happening)

Reply:
Seems like intentional design. Incentives are not there yet but idea is to have some type of "keepers" calling `earn` when necessary.

Reply:
Agree, also some checks on line 85 like `require(strategy !== 0x0,"!bad_strategy")`

And a check that the amount and balance are correct? not sure.